packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -159,12 +192,21 @@ export class SummarizerNode implements IRootSummarizerNode {
 			}
 		}
 
-		const result = await this.summarizeInternalFn(fullTree, true, telemetryContext);
-		this.wipLocalPaths = { localPath: EscapedPath.create(result.id) };
-		if (result.pathPartsForChildren !== undefined) {
-			this.wipLocalPaths.additionalPath = EscapedPath.createAndConcat(
-				result.pathPartsForChildren,
-			);
+		const result = await scheduleIdleTask(async (deadline: IdleDeadline) => {
+			if (deadline.didTimeout) {
```

Comment:
When `didTimeout` is false, maybe we can log `timeRemaining` to give us an idea of how whether the 5 seconds timeout is too high. The log will be too noisy though.